pystiche_papers/sanakoyeu_et_al_2018/_data.py

```diff
@@ -0,0 +1,129 @@
+from typing import Any, Dict, Optional, Sized, cast
+
+import torch
+from torch.utils.data import DataLoader, Dataset, Sampler
+
+from pystiche.data import ImageFolderDataset
+from pystiche.image.transforms import ComposedTransform, Transform, ValidRandomCrop
+from pystiche.image.transforms import functional as F
+from pystiche.image.utils import extract_image_size
+
+from ..data.utils import FiniteCycleBatchSampler
+from ..utils import OptionalGrayscaleToFakegrayscale
+
+
+def image_transform(edge_size: int = 768,) -> ComposedTransform:
+    class OptionalRescale(Transform):
+        # https://github.com/pmeier/adaptive-style-transfer/blob/07a3b3fcb2eeed2bf9a22a9de59c0aea7de44181/prepare_dataset.py#L49-L68
+        def __init__(
+            self, interpolation_mode: str = "bilinear",
+        ):
+            super().__init__()
+            self.interpolation_mode = interpolation_mode
+
+        def forward(
+            self, image: torch.Tensor, maximal_size: int = 1800, minimal_size: int = 800
```

Comment:
Shouldn't `maximal_size` and `minimal_size` instance parameter or do they change in between calls?